README.md

```diff
@@ -132,105 +130,119 @@ CeresDB is a Schema-less time-series database, so creating table schema ahead of
-        Assert.assertEquals(3, wr.mapOr(0, WriteOk::getSuccess).intValue());
-        Assert.assertEquals(0, wr.getOk().getFailed());
-        Assert.assertEquals(0, wr.mapOr(-1, WriteOk::getFailed).intValue());
+final Result<WriteOk, Err> writeResult = wf.get();
+Assert.assertTrue(writeResult.isOk());
+// `Result` class referenced the Rust language practice, provides rich functions (such as mapXXX, andThen) transforming the result value to improve programming efficiency. You can refer to the API docs for detail usage.
+Assert.assertEquals(3, writeResult.getOk().getSuccess());
+Assert.assertEquals(3, writeResult.mapOr(0, WriteOk::getSuccess).intValue());
+Assert.assertEquals(0, writeResult.getOk().getFailed());
+Assert.assertEquals(0, writeResult.mapOr(-1, WriteOk::getFailed).intValue());
 ```
 See [write](docs/write.md)
 
 ## Query data example
 ```java
-final QueryRequest queryRequest = QueryRequest.newBuilder()
-        .forMetrics("machine_metric") // table name is optional. If not provided, SQL parser will parse the `ql` to get the table name and do the routing automaticly
-        .ql("select timestamp, cpu, mem from machine_metric") //
+final SqlQueryRequest queryRequest = SqlQueryRequest.newBuilder()
+        .forTables("machine_table") // table name is optional. If not provided, SQL parser will parse the `ssql` to get the table name and do the routing automaticly
+        .sql("select * from machine_table where ts = %d", t0) //
         .build();
-final CompletableFuture<Result<QueryOk, Err>> qf = client.query(queryRequest);
+final CompletableFuture<Result<SqlQueryOk, Err>> qf = client.sqlQuery(queryRequest);
 // here the `future.get` is just for demonstration, a better async programming practice would be using the CompletableFuture API
-final Result<QueryOk, Err> qr = qf.get();
+final Result<SqlQueryOk, Err> queryResult = qf.get();
+
+Assert.assertTrue(queryResult.isOk());
 
-        Assert.assertTrue(qr.isOk());
+final SqlQueryOk queryOk = queryResult.getOk();
+Assert.assertEquals(1, queryOk.getRowCount());
 
-final QueryOk queryOk = qr.getOk();
+// get rows as list
+final List<Row> rows = queryOk.getRowList();
+Assert.assertEquals(t0, rows.get(0).getColumnValue("ts").getTimestamp());
+Assert.assertEquals("Singapore", rows.get(0).getColumnValue("city").getString());
+Assert.assertEquals("10.0.0.1", rows.get(0).getColumnValue("ip").getString());
+Assert.assertEquals(0.23, rows.get(0).getColumnValue("cpu").getDouble(), 0.0000001);
+Assert.assertEquals(0.55, rows.get(0).getColumnValue("mem").getDouble(), 0.0000001);
 
-final List<Record> records = queryOk.mapToRecord().collect(Collectors.toList())
+// get rows as stream
+final Stream<Row> rowStream = queryOk.stream();
+rowStream.forEach(row -> System.out.println(row.toString()));
```

Comment:
Use `LOG.info()` replace `System.out.println()`.

Reply:
Keep it simple in the readme?